src/main.cpp

```diff
@@ -61,9 +62,10 @@ int main(int argc, char **argv)
   }
 #else
   LAMMPS *lammps = new LAMMPS(argc,argv,MPI_COMM_WORLD);
+  MPI_Comm world_comm = lammps->world;
   lammps->input->file();
   delete lammps;
 #endif
-  MPI_Barrier(MPI_COMM_WORLD);
```

Comment:
This seems wrong. LAMMPS is initialized with MPI_COMM_WORLD and may have been set up to have multiple partitions. It must wait for **all** partitions to end cleanly (or abort) before it can finalize MPI and exit. If you take only the world communicator of the individual LAMMPS class instance, it will wait for the individual partition to finish, but not for all. That can lead to crashes or undefined behavior if some MPI ranks already have called MPI_Finalize() while others may still want to be communicating.